polysemy-plugin/src/Polysemy/Plugin/Fundep.hs

```diff
@@ -202,8 +215,9 @@ solveFundep (ref, stuff) given _ wanted = do
     let r  = fcRow fc
     case findMatchingEffectIfSingular fc given_finds of
```

Comment:
Potential bug source? Previously, if `findMatchingEffectIfSingular` finds an single, ununifiable effect, then the `Just` branch would get taken, and then `mkWanted` would reject making an equality constraint as the effect is ununifiable.
With my modifications, the ununifiable effect would get rejected in `findMatchingEffectIfSingular`, and the `Nothing` path will get taken instead.

Reply:
Or perhaps that's actually an improvement?